src/faultlog/faultlog_main.cpp

```diff
@@ -25,6 +26,111 @@ using Severity = sdbusplus::xyz::openbmc_project::Logging::server::Entry::Level;
 
 using Binary = std::vector<uint8_t>;
 using PropVariant = sdbusplus::utility::dedup_variant_t<Binary>;
+
+/** @brief Helper method to create faultlog pel
+ *
+ *  @param[in] bus - D-Bus to attach to
+ *  @param[in] guardRecords - hardware isolated records to parse
+ *  @param[in] hostPowerOn - flag to check if called during host IPL
+ */
+void createNagPel(sdbusplus::bus::bus& bus,
+                  const GuardRecords& unresolvedRecords, bool hostPowerOn)
+{
+    int guardCount = GuardWithEidRecords::getCount(unresolvedRecords);
+    int manualGuardCount = GuardWithoutEidRecords::getCount(unresolvedRecords);
+    int unresolvedPelsCount = UnresolvedPELs::getCount(bus, hostPowerOn);
+    int deconfigCount = DeconfigRecords::getCount();
+
+    if ((guardCount > 0) || (manualGuardCount > 0) ||
+        (unresolvedPelsCount > 0) || (deconfigCount > 0))
+    {
+        std::unordered_map<std::string, std::string> data;
+        data.emplace("GUARD_WITH_ASSOC_ERROR_COUNT",
+                     std::to_string(guardCount));
+        data.emplace("GUARD_WITH_NO_ASSOC_ERROR_COUNT",
+                     std::to_string(manualGuardCount));
+        data.emplace("UNRESOLVED_PEL_WITH_DECONFIG_BIT_COUNT",
+                     std::to_string(unresolvedPelsCount));
+        data.emplace("DECONFIG_RECORD_COUNT", std::to_string(deconfigCount));
+
+        auto method = bus.new_method_call(
+            "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+            "xyz.openbmc_project.Logging.Create", "Create");
+        method.append("org.open_power.Faultlog.Error.DeconfiguredHW",
+                      Severity::Warning, data);
+        auto reply = method.call();
+        if (reply.is_method_error())
+        {
+            lg2::error("Error in calling D-Bus method to create PEL");
+        }
+        lg2::info("faultlog {GUARD_COUNT}, {MAN_GUARD_COUNT}, "
+                  "{DECONFIG_COUNT} , {PEL_COUNT} ",
+                  "GUARD_COUNT", guardCount, "MAN_GUARD_COUNT",
+                  manualGuardCount, "DECONFIG_COUNT", deconfigCount,
+                  "PEL_COUNT", unresolvedPelsCount);
+    }
+    else
+    {
+        lg2::info("There are no pending service actions ignoring "
+                  "creating fautlog pel");
+    }
+}
+
+/** @brief Callback method for boot progress property change
+ *
+ *  @param[in] bus - D-Bus to attach to
+ *  @param[in] unresolvedRecords - hardware isolated records to parse
+ *  @param[in] hostPowerOn - flag to check if called during host IPL
+ *  @param[in] msg - property change D-Bus message
```

Comment:
If we are not changing the property of msg object then a const reference is better I guess